metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/SensorParserConfigService.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.rest.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.metron.common.configuration.ConfigurationType;
+import org.apache.metron.common.configuration.ConfigurationsUtils;
+import org.apache.metron.common.configuration.SensorParserConfig;
+import org.apache.metron.parsers.interfaces.MessageParser;
+import org.apache.metron.rest.model.ParseMessageRequest;
+import org.apache.metron.rest.model.SensorParserConfigVersion;
+import org.apache.metron.rest.repository.SensorParserConfigVersionRepository;
+import org.apache.zookeeper.KeeperException;
+import org.json.simple.JSONObject;
+import org.reflections.Reflections;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Service
+public class SensorParserConfigService {
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  private CuratorFramework client;
+
+  @Autowired
+  public void setClient(CuratorFramework client) {
+    this.client = client;
+  }
+
+  @Autowired
+  private GrokService grokService;
```

Comment:
Why do we have a `GrokService` in a more general `SensorParserConfigService`?  Is there something coupled between grok and the general parser?

Half of this class seems grok specific and the rest is generic to `BasicParser`, so I'm wondering if this could be abstracted differently.